src/v/cloud_storage/topic_manifest.cc

```diff
@@ -12,21 +12,207 @@
 
 #include "bytes/iobuf_istreambuf.h"
 #include "bytes/iobuf_ostreambuf.h"
+#include "cloud_storage/logger.h"
 #include "cloud_storage/types.h"
+#include "cluster/types.h"
 #include "hashing/xx.h"
 #include "json/writer.h"
 #include "model/timestamp.h"
 
 #include <seastar/core/coroutine.hh>
+#include <seastar/core/sstring.hh>
 
 #include <boost/lexical_cast.hpp>
 #include <fmt/ostream.h>
+#include <rapidjson/encodings.h>
+#include <rapidjson/error/en.h>
 #include <rapidjson/istreamwrapper.h>
 #include <rapidjson/ostreamwrapper.h>
+#include <rapidjson/reader.h>
 
 #include <chrono>
+#include <optional>
+#include <stdexcept>
+#include <string_view>
 
 namespace cloud_storage {
+
+struct topic_manifest_handler
+  : public rapidjson::
+      BaseReaderHandler<rapidjson::UTF8<>, topic_manifest_handler> {
+    using key_string = ss::basic_sstring<char, uint32_t, 31>;
+    bool StartObject() {
+        switch (_state) {
+        case expectManifestStart:
+            _state = expectKey;
+            return true;
+        case expectKey:
+        case expectValue:
+            return false;
+        }
+    }
+
+    bool Key(const char* str, rapidjson::SizeType length, bool copy) {
+        switch (_state) {
+        case expectKey:
+            _key = key_string(str, length);
+            _state = expectValue;
+            return true;
+        case expectManifestStart:
+        case expectValue:
+            return false;
+        }
+    }
+
+    bool String(const char* str, rapidjson::SizeType length, bool copy) {
+        std::string_view sv(str, length);
+        switch (_state) {
+        case expectValue:
+            if ("namespace" == _key) {
+                _namespace = model::ns(ss::sstring(sv));
+            } else if ("topic" == _key) {
+                _topic = model::topic(ss::sstring(sv));
+            } else if ("compression" == _key) {
+                _properties.compression
+                  = boost::lexical_cast<model::compression>(sv);
```

Comment:
Did you consider my previous comment about changing the behaviour here to return a nicer error message through the code path with "Failed to parse topic manifest" if the `lexical_cast` fails? Or perhaps catching `bad_lexical_cast` at that point?

Reply:
thanks for pointing out! fixed!